src/cross_staff_offset.lua

```diff
@@ -2,26 +2,43 @@ function plugindef()
     finaleplugin.RequireSelection = true
     finaleplugin.Copyright = "CC0 https://creativecommons.org/publicdomain/zero/1.0/"
     finaleplugin.Author = "Carl Vine"
-    finaleplugin.AuthorURL = "http://carlvine.com"
-    finaleplugin.Version = "v1.21"
-    finaleplugin.Date = "2022/05/23"   
+    finaleplugin.AuthorURL = "http://carlvine.com/?cv=lua"
+    finaleplugin.Version = "v1.26"
+    finaleplugin.Date = "2022/07/09"
+    finaleplugin.AdditionalMenuOptions = [[  CrossStaff Offset No Dialog  ]]
+    finaleplugin.AdditionalUndoText = [[     CrossStaff Offset No Dialog  ]]
+    finaleplugin.AdditionalPrefixes = [[     cross_staff_nodialog = true  ]]
+    finaleplugin.AdditionalDescriptions = [[ Offset horizontal position of cross-staff note entries - NO DIALOG ]]
     finaleplugin.Notes = [[
-        When creating cross-staff notes using the option-downarrow shortcut, the stems of 'crossed' notes
-        are reversed (on the wrong side of the notehead) and so appear too far to the right (if shifting downwards)
-        by the width of one notehead (typically 24EVPU). This script lets you set a horizontal offset for all cross-staff
-        notes in the currently selected region, with a different offset for non-crossed notes. For crossing to the
-        staff below use (-24,0) or (-12,12). For crossing to the staff above use (24,0) or (12,-12). Also specify
-        which layer number (1-4) or "all layers" (0). (This also offers a simple way to reset the horizontal offset
-        of all notes in the selection to zero).
-
-        If you invoke the plugin with Option key (macOS) or Shift key pressed, the script uses the last settings
-        from the dialog.
-    ]]
-    return "CrossStaff Offset", "CrossStaff Offset", "Offset horizontal position of cross-staff note entries"
+        When creating cross-staff notes using the option-downarrow shortcut, the stems of 
+        'crossed' notes are reversed (on the wrong side of the notehead) and appear too far 
+        to the right (if shifting downwards) by the width of one notehead, typically 24EVPU. 
+        This script lets you set a horizontal offset for all cross-staff notes in the 
+        selected region, with a different offset for non-crossed notes.  
+        For crossing to the staff below use (-24,0) or (-12,12).  
+        For crossing to the staff above use (24,0) or (12,-12).  
+        Also specify which layer number to act upon (1-4) or "all layers" (0). 
+        (This also offers a simple way to reset the horizontal offset of all notes in the selection to zero).
+    
+        Under RGPLua (version 0.62+) the script adds an extra menu item that repeats the last 
+        chosen offset without presenting a dialog, for fast changes duplicating the last settings. 
+]]
+   return "CrossStaff Offset", "CrossStaff Offset", "Offset horizontal position of cross-staff note entries"
 end
```

Comment:
The menu option that opens a dialog should include "..." at the end. (But only the menu string, not the undo string.)

Reply:
Thanks as always for your great advice Robert. All taken on board, and saving window position in user settings will be excellent. But I'm having trouble with dialog:RestorePosition() in a modal dialog. I'll submit the current script revision as work-in-progress, with a dummy `test_txt` variable to show behaviour. Position not saving.

(PS: `transpose_by_step` is my favourite transposition tool).

Reply:
2nd re-try to follow, using modeless dialog, is more promising, but the sequencing isn't right yet.